frontend/controller/backend.js

```diff
@@ -23,70 +24,34 @@ function signJSON (json, keypair) {
   }, json)
 }
 
-const contractSubscriptions = {}
-let serverSocket
-
-export function createWebSocket (url: string, options: Object): Promise<Object> {
-  return new Promise((resolve, reject) => {
-    serverSocket = pubsub({
-      url,
-      options,
-      handlers: {
-        open: () => {
-          console.log('websocket connection opened!')
-          resolve(serverSocket)
-        },
-        error: err => {
-          console.log('websocket error:', err.message, err)
-          reject(err)
-        },
-        data: msg => {
-          // TODO: place us in unrecoverable state (see state.js error handling TODOs)
-          if (!msg.data) throw new Error('malformed message: ' + JSON.stringify(msg))
-          switch (msg.type) {
-            case RESPONSE_TYPE.ENTRY:
-              // We MUST use 'state/enqueueHandleEvent' here to ensure handleEvent()
-              // is called AFTER any currently-running calls to syncContractWithServer().
-              // Calling via SBP also makes it simple to implement 'test/backend.js'
-              sbp('state/enqueueHandleEvent', GIMessage.deserialize(msg.data))
-              break
-            case RESPONSE_TYPE.SUB:
-            case RESPONSE_TYPE.UNSUB:
-            case RESPONSE_TYPE.PUB: // .PUB can be used to send ephemeral messages outside of any contract logs
-              console.debug(`NOTE: ignoring websocket event ${msg.type} in room:`, msg.data)
-              break
-            default:
-              console.error('SOCKET UNHANDLED EVENT!', msg) // TODO: this
-          }
-        }
+export function createGIPubSubClient (url: string, options: Object): Object {
+  return createClient(url, {
+    ...options,
+    messageHandlers: {
+      [RESPONSE_TYPE.ENTRY] (msg) {
```

Comment:
There is already a `RESPONSE_TYPE` defined in `~/shared/pubsub.js`. We should not have two `RESPONSE_TYPE`'s, that is very confusing.

So, one of those `RESPONSE_TYPE`s should be deleted, and everything affected by that fixed, and then I think `NOTIFICATION_TYPE` should be used here..?

Reply:
Good catch!